src/Cli/Microsoft.TemplateEngine.Cli/Commands/AliasAssignmentCoordinator.cs

```diff
@@ -154,6 +154,7 @@ internal static IReadOnlyList<(CliTemplateParameter Parameter, IReadOnlySet<stri
             if (!isAliasTaken(qualifiedShortName))
             {
                 aliases.Add(qualifiedShortName);
+                takenAliases.Add(qualifiedShortName);
```

Comment:
No doubt this is a correctly added line, however not sure I understand why the code flows here. GetFreeName above should process this case.

Reply:
Misunderstood the logic below. Agree that it needs simplification.